impl/src/main/java/org/jboss/weld/logging/BeanLogger.java

```diff
@@ -519,4 +519,7 @@
     @Message(id = 1569, value = "Cannot inject injection point metadata in a non @Dependent bean: {0}", format = Format.MESSAGE_FORMAT)
     IllegalArgumentException cannotInjectInjectionPointMetadataIntoNonDependent(Object bean);
 
+    @Message(id = 1570, value = "Bean class which has decorators should declare at least one non private constructor: {0}\n  StackTrace:", format = Format.MESSAGE_FORMAT)
```

Comment:
Well, the message should rather clarify that "relaxed construction" feature does not work for decorators. Maybe we should investigate first whether it wouldn't be possible to fix it, ie. make relaxed construction work with decorators?